src/main.py

```diff
@@ -94,6 +94,8 @@ def main(
         or not isinstance(prompt_manager, IPromptManager)
     ):
         raise ValueError(f"Error while initializing the modules.")
+    
+    runtime.set_report_title(f"Analysis of dataset {dataset_path}")
```

Comment:
I'm not sure if the dataset_path will be correct when used via streamlit, can you check it?